catenets/models/torch/snet.py

```diff
@@ -0,0 +1,192 @@
+from typing import Optional
+
+import torch
+
+import catenets.logger as log
+from catenets.models.constants import (
+    DEFAULT_BATCH_SIZE,
+    DEFAULT_LAYERS_OUT,
+    DEFAULT_LAYERS_R,
+    DEFAULT_N_ITER,
+    DEFAULT_N_ITER_PRINT,
+    DEFAULT_NONLIN,
+    DEFAULT_PENALTY_L2,
+    DEFAULT_SEED,
+    DEFAULT_STEP_SIZE,
+    DEFAULT_UNITS_OUT,
+    DEFAULT_UNITS_R_BIG_S,
+    DEFAULT_VAL_SPLIT,
+)
+from catenets.models.torch.base import BaseCATEEstimator, BasicNet
+
+
+class SNet(BaseCATEEstimator):
```

Comment:
This is not SNet -- there's a slight confusion with terminology here. SNet as implemented in the jax code is different from this; here you've implemented an S-learner (this is partially my bad for the naming in the AISTATS paper). It cannot hurt to have this learner implemented, but we should give it a different name so jax and torch version follow the same convention.